### PR TITLE
Key Value mapper implementation and site

### PR DESCRIPTION
**Pull request by chamathabeysinghe**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,72 @@
-# siddhi-map-keyvalue
\ No newline at end of file
+siddhi-map-keyvalue
+======================================
+
+The **siddhi-map-keyvalue extension** is an extension to <a target="_blank" href="https://wso2.github.io/siddhi">Siddhi</a>
+which is used to convert Key-Value map message to/from Siddhi events.
```
Comment (nirmal070125) [inline README.md]: convert and from doesn't go together. Please rephrase.

File: checkstyle-suppressions.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dilini-muthumala) [inline checkstyle-suppressions.xml]: Please fix the check style issues and remove this file.

File: component/src/main/java/org/wso2/extension/siddhi/map/keyvalue/sinkmapper/KeyValueSinkMapper.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+        } else {
+            Object data[] = event.getData();
+            for (int i = 0; i < data.length; i++) {
+                String attributeName = attributeNameArray[i];
+                result.put(attributeName, data[i]);
+            }
+
+        }
+
```
Comment (dilini-muthumala) [inline component/src/main/java/org/wso2/extension/siddhi/map/keyvalue/sinkmapper/KeyValueSinkMapper.java]: Please remove unnecessary line breaks.

File: component/src/test/java/org/wso2/extension/siddhi/map/keyvalue/sinkmapper/KeyValueSinkMapperTestCase.java [hunk 4]
```diff
@@ -0,0 +1,760 @@
...
+                        map.put("symbol", "IBM");
+                        map.put("price", 32.6f);
+                        map.put("volume", 160L);
+                        AssertJUnit.assertEquals(map, msg);
+                        break;
+
+                    default:
+                        log.error("Received more than expected number of events. Expected maximum : 6," +
+                                "Received : " + wso2Count.get());
+                        AssertJUnit.fail();
```
Comment (dilini-muthumala) [inline component/src/test/java/org/wso2/extension/siddhi/map/keyvalue/sinkmapper/KeyValueSinkMapperTestCase.java]: During the code review, it was suggested to specify the reason for the Assertion failure. Could you please fix it?

File: findbugs-exclude.xml [hunk 5]
```diff
@@ -0,0 +1,20 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (dilini-muthumala) [inline findbugs-exclude.xml]: Remove this file.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+    <properties>
+        <siddhi.version>4.0.0-M73</siddhi.version>
+        <log4j.version>1.2.17.wso2v1</log4j.version>
+        <carbon.feature.plugin.version>3.0.0</carbon.feature.plugin.version>
+        <testng.version>6.11</testng.version>
+        <commons.io.version>2.5</commons.io.version>
+        <jacoco.version>0.7.9</jacoco.version>
+
+        <mavan.findbugsplugin.exclude.file>findbugs-exclude.xml</mavan.findbugsplugin.exclude.file>
+        <mavan.checkstyle.suppression.file>checkstyle-suppressions.xml</mavan.checkstyle.suppression.file>
```
Comment (dilini-muthumala) [inline pom.xml]: Please remove these

**Review by dilini-muthumala — CHANGES_REQUESTED** [review]

**Review by dilini-muthumala — APPROVED** [review]

**Pull request merged**